docs/DeveloperGuide.md

```diff
@@ -4,30 +4,274 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<img src = "images/ArchitectureDiagram.png">
 
+The above **Architecture Diagram** provides the high-level design of the application.
+
+### Main components
+
+The application comprises the following components:
+
+* `Main`, which handles program startup and shutdown, and also interactions between other components.
+* `UI`, which handles user input and showing messages to the user.
+* `Storage`, which handles the loading and saving of data upon program startup and shutdown.
+* `Parser`, which converts user input into commands.
+* `Command`, which are executed to modify the data stored in the program.
+* `Event`, which stores the program's data.
+
+### Interactions between components
+
+<img src = "images/ArchitectureSequenceDiagram.png">
+
+The above **Sequence Diagram** shows how the different components of the system interact with one
+another in the scenario when the command `add -e event -t 1200 -v venue` is executed.
+
+### UI component
+
+<img src = "images/UiClassDiagram.png">
+
+The `UI` component comprises an Ui class, as shown in the above **Class Diagram**.
+
+The `UI` does the following:
+
+* Take in command input from the user and pass the input to `Main`.
+* Show output messages from `Command` to the user after command execution.
+
+### Storage component
+
+<img src = "images/StorageClassDiagram.png">
```

Comment:
Might you consider adding the solid black triangles to indicate the directionality of the labels? While I am able to gather intuitively the directionality of most of the labels, I am unsure about that of the `adopts` label.